core/src/test/java/com/linecorp/armeria/common/RequestContextTest.java

```diff
@@ -307,6 +308,83 @@ public void makeContextAwareCompletableFutureWithAsyncChaining() throws Exceptio
         }
     }
 
+    @Test
+    public void minimalCompletionStageUsingCompleteAsync() throws Exception {
+        final RequestContext context = createContext(false);
+        final CompletableFuture<Integer> originalFuture = new CompletableFuture<>();
+        @SuppressWarnings("unchecked")
+        final RequestContextAwareCompletableFuture<Integer> contextAwareFuture =
+                (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
+        final CompletionStage<Integer> completionStage = contextAwareFuture.minimalCompletionStage();
+
+        contextAwareFuture.completeAsync(() -> {
+            throw new RuntimeException();
+        });
+        contextAwareFuture.complete(1);
+
+        assertThat(contextAwareFuture.join()).isEqualTo(1);
+        assertThat(contextAwareFuture.getNow(null)).isEqualTo(1);
+        assertThat(contextAwareFuture.get()).isEqualTo(1);
+        assertThat(completionStage.toCompletableFuture().get()).isEqualTo(1);
+    }
+
+    @Test
+    public void minimalCompletionStageUsingWhenComplete() throws Exception {
+        final RequestContext context = createContext(false);
+        final CompletableFuture<Integer> originalFuture = new CompletableFuture<>();
+        @SuppressWarnings("unchecked")
+        final RequestContextAwareCompletableFuture<Integer> contextAwareFuture =
+                (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
+        final CompletionStage<Integer> completionStage = contextAwareFuture.minimalCompletionStage();
+
+        AtomicInteger atomicInteger = new AtomicInteger();
+        AtomicReference<Throwable> causeCaptor = new AtomicReference<>();
+        completionStage.whenComplete((result, error) -> {
+            if (error != null) {
+                causeCaptor.set(error);
+            } else {
+                atomicInteger.set(result);
+            }
+        });
+        contextAwareFuture.complete(1);
+
+        assertThat(contextAwareFuture.join()).isEqualTo(1);
+        assertThat(contextAwareFuture.getNow(null)).isEqualTo(1);
+        assertThat(contextAwareFuture.get()).isEqualTo(1);
+        assertThat(atomicInteger.get()).isEqualTo(1);
+        assertThat(causeCaptor.get()).isNull();
+    }
+
+    @Test
+    public void makeContextAwareCompletableFutureUsingCompleteAsync() throws Exception {
+        final RequestContext context = createContext(false);
+        final CompletableFuture<String> originalFuture = new CompletableFuture<>();
+        @SuppressWarnings("unchecked")
+        final RequestContextAwareCompletableFuture<String> contextAwareFuture =
+                (RequestContextAwareCompletableFuture) context.makeContextAware(originalFuture);
```

Comment:
nit: `(RequestContextAwareCompletableFuture<String>)`

Reply:
This part requires forced converting to be tested in Java8.

Reply:
I mean you used `(RequestContextAwareCompletableFuture)` to downcast without the parameter type. If we add the type parameter here which is `<String>`, we no longer need `@SuppressWarnings("unchecked")` in line 358. 😄